hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -507,6 +635,19 @@ public FileStatus[] listStatus(final Path f) throws IOException {
           fileSystemId, FSOperationType.LISTSTATUS, true, tracingHeaderFormat,
           listener);
       FileStatus[] result = abfsStore.listStatus(qualifiedPath, tracingContext);
+      if (getAbfsStore().getAbfsConfiguration().getPrefixMode()
+          == PrefixMode.BLOB) {
+        FileStatus renamePendingFileStatus
+            = abfsStore.getRenamePendingFileStatus(result);
+        if (renamePendingFileStatus != null) {
+          RenameAtomicityUtils renameAtomicityUtils =
+              new RenameAtomicityUtils(this,
```

Comment:
Same as above, don't like the idea of passing the entire filesystem instance

Reply:
Its required, as it needs to call FS APIs.